content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+
+<!-- -->
+
+    //These samplers are declared inside an SRG
+    ...
+    Sampler m_dynamicSampler;
+     
+    // Compiled as HLSL SamplerState
+    Sampler m_staticSamplerState
+    {
+        AddressU = Wrap;
+        AddressV = Wrap;
+        AddressW = Wrap;
+        MagFilter = Linear;
+    };
+     
+    //Compiled as HLSL SamplerComparisonState because of the ComparisonFunc declaration.
+    Sampler m_staticSamplerComparisonState
+    {
+        AddressU = Wrap;
+        AddressV = Wrap;
+        AddressW = Wrap;
+        MagFilter = Linear;
+        ComparisonFunc = Less;
+    };
+    ...
+  
+### Functions
+Functions can be declared in classes, global scopes, or inside SRGs. Declaring functions in classes or in the global scope is a standard HLSL feature. Defining functions inside SRGs is unique to AZSL (Of course, because ShaderResourceGroup  is a exclusive feature of AZSL).  
+  
+    ShaderResourceGroup PerObject : BindingPerObject
+    {
+        row_major float3x4 m_modelToWorld;
+        row_major float3x3 m_normalMatrix;
+       
+        float4x4 GetWorldMatrix()
+        {
+            float4x4 modelToWorld = float4x4(
+                float4(1, 0, 0, 0),
+                float4(0, 1, 0, 0),
+                float4(0, 0, 1, 0),
+                float4(0, 0, 0, 1));
+       
+            modelToWorld[0] = m_modelToWorld[0];
+            modelToWorld[1] = m_modelToWorld[1];
+            modelToWorld[2] = m_modelToWorld[2];
+            return modelToWorld;
+        }
+           
+        float3x3 GetNormalMatrix()
+        {
+            return m_normalMatrix;
+        }
+    }
+      
+      
+    VertexOutput MainVS(VertexInput input)
+    {
+        const float4x4 worldMatrix = PerObject::GetWorldMatrix();
+       
+        VertexOutput output;
+        float3 worldPosition = mul(worldMatrix, float4(input.m_position,1)).xyz;
+        output.m_position = mul(GetView_ViewProjectionMatrix(), float4(worldPosition, 1.0));
+       
+        output.m_uv = input.m_uv;
+               
+        output.m_positionToCamera = GetView_WorldPosition() - worldPosition;
+       
+        output.m_normal = mul(PerObject::GetNormalMatrix(), input.m_normal);
+        output.m_tangent = mul(PerObject::GetNormalMatrix(), input.m_tangent);
+       
+        return output;
+    }
+  
+### Structs
+Just like in HLSL, structs can be defined globally, but in AZSL they can also be defined and instantiated inside SRGs. You can only instantiate globally defined structs inside SRGs, rootconstants or inside another struct or class.  
+  
+In general, structs are supported just like in HLSL. struct has no constructors/destructors and as such, do not accept default member initializers.  
+  
+Just like in HLSL, struct has the particularity to be data only (no methods).
```

Comment:
- I think we can tighten the comparison between HLSL and AZSL by making separate lists of the rules that HLSL brings and of the additional rule(s) that AZSL brings. 

- Add space between "root constants", as that's how they are written in [Direct3D docs]( https://docs.microsoft.com/en-us/windows/win32/direct3d12/using-constants-directly-in-the-root-signature). 
```suggestion
AZSL supports structs similarly as in HLSL and adds extended rules. 

Just like in HLSL, the following rules apply for structs:
- Can be defined globally. 
- Contains only data, not methods. 
- Don't have a constructor or destructor. 
- Don't accept default member initializers.

AZSL adds an extended rule: Define and instantiate structs inside SRGS. You can instantiate globally defined structs inside SRGs, root constants, and another struct or class. 
```

Reply:
Fixed.